plugins/analytics/public/onyphe.py

```diff
@@ -0,0 +1,369 @@
+import sys
+import json
+import logging
+import requests
+from requests.utils import quote
+
+from core.analytics import OneShotAnalytics
+from core.errors import ObservableValidationError
+from core.observables import Hostname, Ip
+
+default_types = {
+    "Hostname": Hostname,
+    "Ip": Ip,
+}
+
+if sys.version_info.major == 2:
+    from urlparse import urljoin
+elif sys.version_info.major == 3:
+    from urllib.parse import urljoin
+
+
+"""
+onyphe.client
+~~~~~~~~~~~~~
+
+This module implements the Onyphe API.
+
+:copyright: (c) 2017- by Sebastien Larinier
+"""
+
+class APIError(Exception):
+    """This exception gets raised whenever a non-200 status code was returned by the Onyphe API."""
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return self.value
+
+class Onyphe:
+    """Wrapper around the Onyphe REST
+
+        :param key: The Onyphe API key that can be obtained from your account page (https://www.onyphe.io)
+        :type key: str
+    """
+
+    def __init__(self, api_key, version='v1'):
+        self.api_key = api_key
+        self.base_url = 'https://www.onyphe.io/api/'
+        self.version = version
+        self._session = requests.Session()
+
+        self.methods = {
+            'get': self._session.get,
+            'post': self._session.post,
+        }
+
+    def _choose_url(self, uri):
+        self.url = urljoin(self.base_url, uri)
+
+    def _request(self, method, payload):
+
+        data = None
+
+        try:
+            response = self.methods[method](self.url, params=payload)
+        except:
+            raise APIError('Unable to connect to Onyphe')
+
+        if response.status_code == requests.codes.NOT_FOUND:
+
+            raise APIError('Page Not found %s' % self.url)
+        elif response.status_code == requests.codes.FORBIDDEN:
+            raise APIError('Access Forbidden')
+        elif response.status_code == requests.codes.too_many_requests:
+            raise APIError('Too Many Requests')
+        elif response.status_code != requests.codes.OK:
+            try:
+                error = response.json()['message']
+            except Exception as e:
+                error = 'Invalid API key'
+
+            raise APIError(error)
+        try:
+
+            data = response.json()
+
+        except:
+            raise APIError('Unable to parse JSON')
+
+        return data
+
+    def _prepare_request(self, uri, **kwargs):
+        payload = {
+            'apikey': self.api_key
+        }
+
+        if 'page' in kwargs:
+            payload['page'] = kwargs['page']
+
+        self._choose_url(uri)
+
+        data = self._request('get', payload)
+        if data:
+            return data
+
+    def __search(self,query, endpoint, **kwargs):
```

Comment:
Why double underscore?

Reply:
that is from original https://github.com/sebdraven/pyonyphe/ code